packages/block-library/src/block/edit.js

```diff
@@ -333,6 +340,15 @@ export default function ReusableBlockEdit( {
 					</ToolbarGroup>
 				</BlockControls>
 			) }
+			{ overrides ? (
+				<BlockControls>
+					<ToolbarGroup>
+						<ToolbarButton onClick={ resetOverrides }>
+							{ __( 'Reset overrides' ) }
+						</ToolbarButton>
+					</ToolbarGroup>
+				</BlockControls>
+			) : null }
```

Comment:
Actually, one thing is that selecting the button causes a focus loss due to the way it's no longer rendered, so it should probably have a aria-disabled state instead (one that is focusable).

Reply:
Good catch! Or should we move the focus to the pattern block instead? Not sure which is better atm.

Reply:
Most of the accessibility feedback I see is that focus shouldn't be moved unexpectedly if there's an alternative, so that's why I feel that a disabled state would be the way to go. There's also a lot of precedence for this in the block editor, like the undo/redo buttons that retain focus when they become disabled.

It does introduce the caveat around patterns without overrides, should they have a visible button or should it be disabled? My feeling is that there probably shouldn't be a button at all since it would never become enabled.

Reply:
The problem is that patterns without overrides have the same state as after resetting the overrides 🤔. Unless we keep the previous state somehow, but I think that's more confusing than always showing the disabled button. Am I misunderstanding something here?

Reply:
> The problem is that patterns without overrides have the same state as after resetting the overrides

We do look through the inner blocks to check for any that can have overrides here:
https://github.com/WordPress/gutenberg/blob/f6d29bd18e538ff63a52c7984376df58b5730a15/packages/block-library/src/block/edit.js#L141-L148

So given we do that we could also derive whether the pattern has any override attributes within it at the same time and use that to control visibility of the button. I think you end up with three states:
1. Pattern has inner blocks that can be overriden but has no override state - show the button but disabled
2. Pattern has inner blocks that can be overriden and has override state - show the button enabled
3. None of the pattern's inner blocks can be overriden - don't show the button

Reply:
Oh I see what you mean now! I was only considering the (1) and the (2) states.